skyportal/handlers/api/annotation.py

```diff
@@ -30,26 +51,65 @@ def get(self, annotation_id):
               application/json:
                 schema: Error
         """
-        annotation = Annotation.get_if_accessible_by(
-            annotation_id, self.current_user, raise_if_none=True
-        )
+
+        try:
+            annotation_id = int(annotation_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid (scalar integer) annotation ID. ")
+
+        if associated_resource_type.lower() == "sources":
+            try:
+                annotation = Annotation.get_if_accessible_by(
+                    annotation_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error('Could not find any accessible annotations.')
```

Comment:
Set appropriate status 404

@acrellin Would it make sense to catch AccessError in the errorhandler and do some automated response?